src/solvers/refinement/string_refinement.cpp

```diff
@@ -852,13 +852,11 @@ static optionalt<exprt> get_array(
 /// \return a string
 static std::string string_of_array(const array_exprt &arr)
 {
-  unsigned n;
   if(arr.type().id()!=ID_array)
       return std::string("");
 
   exprt size_expr=to_array_type(arr.type()).size();
-  PRECONDITION(size_expr.id()==ID_constant);
-  to_unsigned_integer(to_constant_expr(size_expr), n);
+  auto n = numeric_cast_v<unsigned>(size_expr);
```

Comment:
Why do you use sometimes unsigned and sometimes sizet for values that denote sizes?